### DISCUSSION TITLE
Issues with IFCConvert 7.0 Output

### DISCUSSION THREAD

**Discussion by rwar33** [IfcOpenShell/IfcOpenShell#2892] [General]
Hi All,

We have had IfcConvert embedded in our product for some time with no issues.  We recently discovered an issue with the input data from one of our regular software sources, SDS2.   They have an IFC export feature that exports their created model in Coordination view 2.0, IFC 2X3 schema.  Previously, this was powered by STEP Tools (https://www.steptools.com/sln/ifc/) and was able to be converted to OBJ successfully. However, they have recently switched to ODA (https://www.opendesign.com/products/ifc-sdk)  and now we are seeing failed geometry in the OBJ output.  I have attached the IFC source file and the output OBJ file from IfcConvert 7.0.

[B1528_Canam.zip](https://github.com/IfcOpenShell/IfcOpenShell/files/11036967/B1528_Canam.zip)

Has anyone experienced this type of geometry failure or worked with the ODA SDK before?  Any insight would be helpful.  Thanks!!

Randy
Eterio Realities

**Comment by aothms** [comment]
Hi Randy,

Sorry for the delay. The issue here is large coordinates:

The model is so far away from the origin that there is no precision left anymore to represent the actual geometric detail. Floating point numbers can be large or small, but they are not evenly spaced over the number line. The larger the value, the larger the difference to the next value.

Your output

```
# File generated by IfcOpenShell v0.7.0-476ab506d
g product-284b72ca-fd10-464f-9539-fe0cf2f85e54-body
s 1
v 313764.577436323 5041668.90333321 44.196
v 313764.577436323 5041668.73747121 44.196
```

![afbeelding](https://user-images.githubusercontent.com/1096535/229051999-82846a47-8df1-4ef5-8b6a-4ac78019d2ea.png)

If you convert with `IfcConvert --building-local-placement`

```
# File generated by IfcOpenShell v0.7.0-476ab506d
g product-284b72ca-fd10-464f-9539-fe0cf2f85e54-body
s 1
v 140.328649999984 11.0049310002533 44.196
v 140.328649999984 10.8390690002533 44.196
```

Take a look at the Y component difference between these two vertices and you will see the effect of the distortion.

![afbeelding](https://user-images.githubusercontent.com/1096535/229051953-9f0fa84c-c749-4d04-b4af-5d62050bbb1c.png)

Note that this is only applicable to OBJ output, because that does not have a separation between placement and vertex coords. The Collada DAE output does have node placement and mesh vertex positions defined separately which in this case means that the large values (placement) don't blend with the small values (vertex coords) into a single number. Collada is anyway recommended because because of this separation you can reuse geometries for identical elements.

So using `--building-local-placement` is the easiest option. This kind of large offsets are typically introduced in the site. It might mean that it becomes harder to overlay multiple models. But I don't think that will be an issue in your case.

**Reply by rwar33** [reply]
We are testing and looking into this now.  It seems like it will be the solution for us.

Thanks for the help Thomas!
